app/controllers/api/v2/hosts_controller.rb

```diff
@@ -42,6 +42,17 @@ class HostsController < V2::BaseController
       add_scoped_search_description_for(Host)
 
       def index
+        if params[:with_facts]
```

Comment:
Oh, this feels wrong. How many dedicated `params` are we going to add in the future?
I can imagine a lot of them.

Reply:
I agree it looks wrong, but what would be a better alternative? a specific endpoint?

Reply:
I could think of two options:
1. extend the API to allow a user to limit the fields in a generic way (`?fields=name,id,facts`)
2. use graphql which has all this build in